train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -59,7 +70,8 @@ private void enforceSpeedLimit() {
 
 	@Override
 	public void setJoystickPosition(int joystickPosition) {
-		this.step = joystickPosition;		
+		this.step = joystickPosition;	
+		timer.schedule(task, 1000L,1000L);
```

Comment:
You have to put this line to TrainControllerImpl constructor, and the tests will be good.

Reply:
Ok, I will make it